packages/x-date-pickers/src/internals/utils/valueManagers.ts

```diff
@@ -0,0 +1,53 @@
+import type { PickerStateValueManager } from '../hooks/usePickerState';
+import type { DateValidationError } from '../hooks/validation/useDateValidation';
+import type { TimeValidationError } from '../hooks/validation/useTimeValidation';
+import type { DateTimeValidationError } from '../hooks/validation/useDateTimeValidation';
+import type { FieldSection, FieldValueManager } from '../hooks/useField';
+import { replaceInvalidDateByNull } from './date-utils';
+import {
+  addPositionPropertiesToSections,
+  createDateStrFromSections,
+  splitFormatIntoSections,
+} from '../hooks/useField/useField.utils';
+
+export const singleItemValueManager: PickerStateValueManager<
+  any,
+  any,
+  DateValidationError | TimeValidationError | DateTimeValidationError
+> = {
+  emptyValue: null,
+  getTodayValue: (utils) => utils.date()!,
+  cleanValue: replaceInvalidDateByNull,
+  areValuesEqual: (utils, a, b) => utils.isEqual(a, b),
+  isSameError: (a, b) => a === b,
```

Comment:
Maybe you can remove the following line (same for time and date-time)
```ts
export const isSameDateError = (a: DateValidationError, b: DateValidationError) => a === b;
```

Reply:
It is still used in the `useDateValidation` (that will be dropped once we remove the legacy pickers I think)